.github/workflows/deps.yml

```diff
@@ -30,11 +30,18 @@ jobs:
       # Run the commands to generate dependencies before and after and compare.
       - name: Compare dependencies
         run: |
-          BEFORE="$(mktemp -d)"
-          AFTER="$(mktemp -d)"
+          set -eux
+          BASE_REF="${GITHUB_BASE_REF:-master}"
+          # Suffixes passed to mktemp must not contain "/".
+          BEFORE_SUFFIX="${BASE_REF//\//_}"
+          AFTER_SUFFIX="${GITHUB_HEAD_REF//\//_}"
+
+          BEFORE="$(mktemp -d --suffix="-${BEFORE_SUFFIX}")"
+          AFTER="$(mktemp -d --suffix="-${AFTER_SUFFIX}")"
```

Comment:
Idea:

How about have one `mktemp -d` and then make two subdirectories of it which are the same as the existing `BEFORE_SUFFIX` and `AFTER_SUFFIX` and then `cd` to the `mktemp` dir before  doing the `diff` so the relative path names are the names of the git refs?  WDYT?

Reply:
That's a good idea. Changed accordingly.

Reply:
I changed the subdir names to `before` and `after` to keep things simple. It avoids having to sanitize the branch names to ensure they are valid linux paths, handling same source and target ref names and other possible issues.